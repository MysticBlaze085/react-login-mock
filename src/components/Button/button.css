.btn {
  width: 100%;
  cursor: pointer;
  margin-top: 20px;
  border: 1px solid #cfcfcf;
  padding: 10px;
  border-radius: 5px;
  background-size: 2rem 2rem;
  transition: all 250ms cubic-bezier(0.4, 0, 0.2, 1) 0s;
}

.loading {
  background-color: #c5c3c3;
  background-image: linear-gradient(
    135deg,
    rgba(0, 0, 0, 0.1) 25%,
    transparent 25%,
    transparent 50%,
    rgba(0, 0, 0, 0.1) 50%,
    rgba(0, 0, 0, 0.1) 75%,
    transparent 75%,
    transparent
  );
  color: #ffffff;
  animation: 2.8s linear 0s infinite normal none running loading;
}

@keyframes loading {
  0% {
    background-position: 0px 0px;
  }
  100% {
    background-position: 2rem 4rem;
  }
}
