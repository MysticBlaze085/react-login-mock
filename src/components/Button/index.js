import React from "react";
import PropTypes from "prop-types";

import "./button.css";

function Button({ title = "", onClick = () => {}, loading = false }) {
  return (
    <button
      className={`btn ${loading ? "loading" : ""}`}
      disabled={loading}
      onClick={onClick}
    >
      {title}
    </button>
  );
}

Button.propTypes = {
  title: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  loading: PropTypes.bool,
};

export default Button;
