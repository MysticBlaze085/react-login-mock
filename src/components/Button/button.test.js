import React from "react";
import { render, cleanup, getByRole } from "@testing-library/react";

import Button from "./index";

test("render button component", () => {
  const { asFragment, getByText, rerender } = render(
    <Button title="Button" onClick={() => {}} />
  );
  expect(asFragment()).toMatchSnapshot();
  expect(getByText("Button")).toBeInTheDocument();

  rerender(<Button title="Button" loading={true} onClick={() => {}} />);
  expect(getByText("Button")).toHaveClass("loading");
  expect(getByText("Button")).toBeDisabled();
});

afterEach(cleanup);
