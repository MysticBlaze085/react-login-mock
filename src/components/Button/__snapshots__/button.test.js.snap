// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render button component 1`] = `
<DocumentFragment>
  <button
    class="btn "
  >
    Button
  </button>
</DocumentFragment>
`;
